torchmetrics/metric.py

```diff
@@ -258,27 +269,55 @@ def sync(
                 only when running in a distributed setting.
             distributed_available: Function to determine if we are running inside a distributed setting
 
-        Returns:
-            cache: A dictionary containing the local metric states. The cache will be empty if sync didn't happen.
         """
+        if self.is_synced and should_sync:
+            raise TorchMetricsUserError("The Metric has already been synced.")
+
```

Comment:
its not clear why we have `should_sync` as an argument to `sync` - isn't the user's intention already to sync? 

additionally, should this raise an exception? or log a warning and then return?

Reply:
`should_sync` is provided for now as compute is relying on it. This is legacy behaviour to be cleaned out.

```py
            with self.sync_context(
                dist_sync_fn=self.dist_sync_fn, should_sync=self._to_sync, should_unsync=self._should_unsync
            ):
                self._computed = compute(*args, **kwargs)
```

I believe we should strongly prevent users to have wrong behaviours by raising an exception.

Reply:
Could you add a comment to the code around this as well?

Reply:
Done !